### PR TITLE
[Resources] Import APIs for subscriptions.json file

### PR DESCRIPTION
**Pull request by Tiano2017**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
         >>>>the change is to add APIs that already exist in an existing API version
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
      >>>>The service changes are already in production 
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
      >>>> Sooner the better
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [x] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by markcowl** [comment]
@Tiano2017 This requires an ARM review, and you must fix avocado failures

**Comment by Tiano2017** [comment]
Hi @markcowl, the Avocado failures seem to be false alarm. The examples are referenced from the new subscriptions.json file. Besides that, I'll work on getting ARM review.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json [hunk 1]
```diff
@@ -0,0 +1,730 @@
...
+        "description": "This operation provides all the locations that are available for resource providers; however, each resource provider may support a subset of this list.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "includeExtendedLocations",
```
Comment (RamyasreeChakka) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:78]: LGTM. Signing off from API review side.

**Comment by Tiano2017** [comment]
Hi @markcowl, can you help take another look? I got the ARM API sign off. The Avocado failures seem to be false alarm. The examples are being referenced from the new subscriptions.json file.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json [hunk 2]
```diff
@@ -0,0 +1,730 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Resources/checkResourceName": {
+      "post": {
+        "tags": [
+          "Tenants"
+        ],
+        "operationId": "checkResourceName",
```
Comment (markcowl) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:202]: OperationId must be of the form "OperationGroup_Operation", like "Resources_CheckName", or "Tenants_CheckResourceName"

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json [hunk 3]
```diff
@@ -0,0 +1,730 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns an array of tenants.",
+            "schema": {
+              "$ref": "#/definitions/TenantListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
```
Comment (markcowl) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:187]: Not positive why these are not registering with Avocado, but my guess is that it is the period in the example names.  This will need to be fixed before check-in.

Comment (markcowl) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:187]: same for the others.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json [hunk 4]
```diff
@@ -0,0 +1,730 @@
...
+        "homeLocation": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The home location of an edge zone."
+        }
+      }
+    },
+    "Location": {
+      "properties": {
+        "id": {
```
Comment (markcowl) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:325]: Is Location a resource?  If so, it should extend one of the common resource types.  Also, does it have a 'systemData' property?

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json [hunk 5]
```diff
@@ -0,0 +1,730 @@
...
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Location"
+          },
+          "description": "An array of locations."
+        }
+      },
+      "description": "Location list operation response."
```
Comment (markcowl) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:380]: next link?

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json [hunk 6]
```diff
@@ -0,0 +1,730 @@
...
+    },
+    "Subscription": {
+      "properties": {
+        "id": {
```
Comment (markcowl) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:384]: Is this a resource (the properties indicate it may be)?  If so, it shoudl extend one of the common resource types

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json [hunk 7]
```diff
@@ -0,0 +1,730 @@
...
+          ],
+          "x-ms-enum": {
+            "name": "ResourceNameStatus",
+            "modelAsString": true
+          }
+        }
+      },
+      "description": "Resource Name valid if not a reserved word, does not contain a reserved word and does not start with a reserved word"
+    },
+    "ErrorResponse": {
```
Comment (markcowl) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:633]: You should use the common-types error definition, which matches this one.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json [hunk 8]
```diff
@@ -0,0 +1,730 @@
...
+            "schema": {
+              "$ref": "#/definitions/LocationListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-examples": {
+          "Get locations with a subscription id": {
```
Comment (markcowl) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:98]: The example keys may need not to include spaces

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json [hunk 9]
```diff
@@ -0,0 +1,730 @@
...
+        "responses": {
+          "200": {
+            "description": "OK - Returns an array of subscriptions.",
+            "schema": {
+              "$ref": "#/definitions/SubscriptionListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get all subscriptions.": {
```
Comment (markcowl) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:158]: remove spaces and period

**Review by markcowl — CHANGES_REQUESTED** [review]
Note: you *must* clean up the avocado errors before check-in, you *should* look at the other failures as well.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json (hunk 9 above)
Comment (Tiano2017) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:158]: updated.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json (hunk 8 above)
Comment (Tiano2017) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:98]: updated.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json (hunk 3 above)
Comment (Tiano2017) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:187]: updated.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json (hunk 2 above)
Comment (Tiano2017) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:202]: These are copied from the previous version. Can we leave them? BTW, the first commit of this PR was a copy from previous version, and the subsequent commits contain the actual changes.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json (hunk 4 above)
Comment (Tiano2017) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:325]: Location is not a resource as of today, although we do have this "id" property which put here for future extensibility.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json (hunk 5 above)
Comment (Tiano2017) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:380]: unfortunately we don't support paging on this API yet.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json (hunk 6 above)
Comment (Tiano2017) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:384]: The subscription on this API is not a resource. The "id" property is only to identify the subscription. I know Commerce RP modeled subscription as a resource and provides separate APIs under their namespace.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json (hunk 7 above)
Comment (Tiano2017) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:633]: updated

**Comment by Tiano2017** [comment]
@markcowl I updated the x-ms-examples, but still getting the avocado failures..

**Comment by Tiano2017** [comment]
@markcowl  Also, regarding the error on breaking change. The property is already added in service side a few versions back, so what would be the right process here. Should I still create a new swagger version to include that change? That doesn't seem to be meaningful.

File: specification/common-types/resource-management/v1/types.json [hunk 10]
```diff
@@ -120,6 +120,13 @@
           ],
           "description": "The geo-location where the resource lives"
         },
+        "extendedLocation": {
```
Comment (markcowl) [inline specification/common-types/resource-management/v1/types.json:123]: This changes every single resource from every RP in every api-version that references ResourceModelWithAllowedPropertySet - in other words, it is potentially a massive breaking change across all RPs.  We cannot simply add a new property to the resource definitions in common types.

We should likely add a new resource type, or create a new version of common-types.

Also, this modelling is strange - you are creating a new inline schema definition that extends ExtendedLocation, rather than just referencing the ExtendedLocation schema.  When you do model this in a new type, you should just reference the schema for ExtendedLocation

File: specification/common-types/resource-management/v2/types.json [hunk 11]
```diff
@@ -120,6 +120,13 @@
           ],
           "description": "The geo-location where the resource lives"
         },
+        "extendedLocation": {
```
Comment (markcowl) [inline specification/common-types/resource-management/v2/types.json:123]: same comments.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/resources.json [hunk 12]
```diff
@@ -5381,6 +5381,26 @@
       ],
       "description": "Resource information."
     },
+    "ExtendedLocation": {
```
Comment (markcowl) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/resources.json:5384]: Why did you add this in common-types and then not reference it?

File: specification/common-types/resource-management/v2/types.json [hunk 13]
```diff
@@ -179,6 +186,26 @@
       },
       "x-ms-azure-resource": true
     },
+    "ExtendedLocation": {
+      "description": "The resource model definition representing extended location",
+      "properties": {
+        "type": {
```
Comment (markcowl) [inline specification/common-types/resource-management/v2/types.json:192]: what is the mutability of this property?  Can it be changed on update, or only on create?  Please appropriately mark with x-ms-mutability.  This applies to 'name' as well.

File: specification/common-types/resource-management/v1/types.json (hunk 10 above)
Comment (markcowl) [reply specification/common-types/resource-management/v1/types.json:123]: What is the mutability?

File: specification/common-types/resource-management/v1/types.json [hunk 14]
```diff
@@ -179,6 +186,26 @@
       },
       "x-ms-azure-resource": true
     },
+    "ExtendedLocation": {
+      "description": "The resource model definition representing extended location",
+      "properties": {
+        "type": {
```
Comment (markcowl) [inline specification/common-types/resource-management/v1/types.json:192]: need to specify the mutability of these properties.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/resources.json [hunk 15]
```diff
@@ -5954,6 +5974,10 @@
           "type": "string",
           "description": "Resource location"
         },
+        "extendedLocation": {
```
Comment (markcowl) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/resources.json:5977]: breaking change

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json (hunk 2 above)
Comment (markcowl) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/subscriptions.json:232]: yes, this will just add it as a top level client method

**Review by markcowl — CHANGES_REQUESTED** [review]

File: specification/resources/resource-manager/Microsoft.Resources/stable/2021-01-01/resources.json [hunk 16]
```diff
@@ -0,0 +1,6392 @@
...
+          "type": "string",
+          "description": "Resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location"
+        },
+        "extendedLocation": {
+          "$ref": "#/definitions/ExtendedLocation",
+          "description": "Resource extended location."
```
Comment (markcowl) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2021-01-01/resources.json:5979]: since this is new, we should have the correct x-ms-mutability (location should as well, but this would technically be a breaking change).  If this is determined by the service, it should be set as readOnly.  If it can be set on creation only, it should be "read", "create".  If it is settable on create and update, then you can leave it as is.

**Review by markcowl — CHANGES_REQUESTED** [review]

File: specification/resources/resource-manager/Microsoft.Resources/stable/2021-01-01/resources.json (hunk 16 above)
Comment (Tiano2017) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2021-01-01/resources.json:5983]: udpated.

**Review by markcowl — APPROVED** [review]
